app/src/gecko/java/org/mozilla/focus/web/GeckoViewPrompt.java

```diff
@@ -0,0 +1,888 @@
+            if (picker == null) {
+                picker = new DatePicker(builder.getContext());
+            }
+            picker.init(cal.get(Calendar.YEAR), cal.get(Calendar.MONTH),
+                    cal.get(Calendar.DAY_OF_MONTH), /* listener */ null);
+            if (minDate != null) {
+                picker.setMinDate(minDate.getTime());
+            }
+            if (maxDate != null) {
+                picker.setMaxDate(maxDate.getTime());
+            }
+            datePicker = picker;
+        } else {
+            datePicker = null;
+        }
+
+        final TimePicker timePicker;
+        if (type == DATETIME_TYPE_TIME || type == DATETIME_TYPE_DATETIME_LOCAL) {
+            final int resId = builder.getContext().getResources().getIdentifier(
+                    "time_picker_dialog", "layout", "android");
+            TimePicker picker = null;
+            if (resId != 0) {
+                try {
+                    picker = (TimePicker) inflater.inflate(resId, /* root */ null);
+                } catch (final ClassCastException | InflateException e) {
+                }
+            }
+            if (picker == null) {
+                picker = new TimePicker(builder.getContext());
+            }
+            setTimePickerTime(picker, cal);
+            picker.setIs24HourView(DateFormat.is24HourFormat(builder.getContext()));
+            timePicker = picker;
+        } else {
+            timePicker = null;
+        }
+
+        final LinearLayout container = addStandardLayout(builder, title, /* msg */ null);
+        container.setPadding(/* left */ 0, /* top */ 0, /* right */ 0, /* bottom */ 0);
+        if (datePicker != null) {
+            container.addView(datePicker);
+        }
+        if (timePicker != null) {
+            container.addView(timePicker);
+        }
+
+        final DialogInterface.OnClickListener listener =
+                new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(final DialogInterface dialog, final int which) {
+                        if (which == DialogInterface.BUTTON_NEUTRAL) {
+                            // Clear
+                            callback.confirm("");
+                            return;
+                        }
+                        if (datePicker != null) {
+                            cal.set(datePicker.getYear(), datePicker.getMonth(),
+                                    datePicker.getDayOfMonth());
+                        }
+                        if (timePicker != null) {
+                            setCalendarTime(cal, timePicker);
+                        }
+                        callback.confirm(formatter.format(cal.getTime()));
+                    }
+                };
+        builder.setNegativeButton(android.R.string.cancel, /* listener */ null)
+                .setNeutralButton(R.string.gv_prompt_clear, listener)
+                .setPositiveButton(android.R.string.ok, listener);
```

Comment:
\[PMD\] INFO: The method 'promptForFile' has a Standard Cyclomatic Complexity of 11\.